sherpa/data.py

```diff
@@ -444,31 +454,43 @@ def mask(self):
 
     @mask.setter
     def mask(self, val):
-        if (val is True) or (val is False):
-            self._mask = val
-        # if val is of type np.bool_ and True, it failed the previous test because
-        # "is True" compares with Python "True" singelton.
-        # Yet, we do not want to allow arbitrary values that evaluate as True.
+        # Special case the numpy.ma.nomask setting (which evaluates to False
+        # but is not a bool).
+        #
+        if val is None:
+            raise DataErr('ismask')
+
+        elif not numpy.isscalar(val):
+            # We could catch an error if the value can not be converted,
+            # but wat values can not be converted to a bool?
+            self._mask = numpy.asarray(val, bool)
+
         elif val is numpy.ma.nomask:
             self._mask = True
-        elif numpy.isscalar(val) and isinstance(val, numpy.bool_):
+
+        elif (val is True) or (val is False):
+            self._mask = val
+
+        elif isinstance(val, numpy.bool_):
```

Comment:
I suggest we keep this comment, it's not obvious to me and I wrote that code originally!
```suggestion
        # if val is of type np.bool_ and True, it failed the previous test because
        # "is True" compares with Python "True" singelton.
        # Yet, we do not want to allow arbitrary values that evaluate as True.
        elif isinstance(val, numpy.bool_):
```

Reply:
In #1477 I've re-written this so there's introductory text to explain this (at least, in my current branch which hasn't been pushed yet).